### ISSUE TITLE
v4.0.0-beta.9 custom action handlers

### ISSUE THREAD

**Issue by KaiKloepfer** [auth0/nextjs-auth0#1837]
### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/nextjs-auth0#readme), [Examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/nextjs-auth0/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

In v3, we were able to define custom handlers for actions such as login, logout, signup, etc. We currently use these to track analytics, events and manipulate the url.

For example:
```typescript
import { Handler } from "@auth0/nextjs-auth0/dist/handlers/router-helpers";
import { handleAuth, handleLogin } from "@auth0/nextjs-auth0";
import {
  AppRouteHandlerFnContext,
  LoginOptions,
} from "@auth0/nextjs-auth0/dist/handlers";
import { NextRequest } from "next/server";

const signup: Handler = async (req, ctx) => {
  const options: LoginOptions = {
    authorizationParams: {
      screen_hint: "signup",
    },
  };

  await trackSimpleEvent({ event: "Started Auth0 Signup" });
  return handleLogin(
    req as NextRequest,
    ctx as AppRouteHandlerFnContext,
    options
  );
};

const login: Handler = async (req, ctx) => {
  const options: LoginOptions = {};

  if (req.url) {
    const url = new URL(req.url);
    const loginHint = url.searchParams.get("login_hint");
    if (loginHint) {
      options.authorizationParams = {
        login_hint: loginHint,
      };
    }
  }

  await trackSimpleEvent({ event: "Started Auth0 Login" });
  return handleLogin(
    req as NextRequest,
    ctx as AppRouteHandlerFnContext,
    options
  );
};

export const GET = handleAuth({
  signup,
  login,
});
```

### Describe the ideal solution

Support a similar workflow in v4

**Comment by guabu** [comment]
Hey @KaiKloepfer 👋 You should be able to pass custom authorization parameters dynamically by appending them as query parameters to the `/auth/login` route or statically by defining them in the `authorizationParameters` option when instantiating the client. There's a section in the docs that has some examples of both approaches: https://github.com/auth0/nextjs-auth0/tree/v4?tab=readme-ov-file#passing-authorization-parameters

For tracking events, you might be able to do this via the middleware by checking the path. For example:

```ts
if (request.nextUrl.pathname === '/auth/login') {
  await trackSimpleEvent({ event: "Started Auth0 Login" })
}
```

**Comment by nandan-bhat** [comment]
Closing this issue as it has been resolved in `4.0.0-beta.11`.
